### DISCUSSION TITLE
How accurate is the Fermi level printed in detailed.out?

### DISCUSSION THREAD

**Discussion by achyuttt** [dftbplus/dftbplus#1506] [General]
As a new DFTB+ user, I have been facing lots of problems on validating the DFTB+ results. I wanted to make sure that my calculations are correct using DFTB+ but I couldn't find a relevant literature for the work I am doing.

I just wanted to confirm how accurate will the fermi level printed in detailed.out file is.
I am using that value for the calculation of bandgap but for the system that should have fermi level in between band gap, I am getting Fermi level somewhere it shouldn't be. Is there a requirement for SCC Tolerance to be to accurately get the Fermi level? or are there any parameters to be changed in the input file to get the accurate Fermi level?

**Comment by aradi** [comment]
If you have a large band gap system, the position of the Fermi-level at low temperatures is numerically ill defined. Tightening the SCC criterion won't help here at all. In case of fractionally filled systems (metalic systems) the position of the Fermi level is well defined. Tightening the SCC criterion there would give a more accurate position.

**Reply by achyuttt** [reply]
For a semiconductor system ( band gap usually below 3 eV ), how accurate will it be?
How can I verify if the Fermi Level that I am getting in the detailed.out file is numerically correct for my system?
I am trying to find band gap for the Covalent Organic Framework from density of states and I use the Fermi Level printed out in the detailed.out file for the calculation of band gap. 
I just wanted to be sure that if Fermi level that I get from DOS calculation is correct enough, my calculated band gap will be correct but I couldn't be assured of the fermi level given by DFTB+ for such semiconductor systems.

**Reply by aradi** [reply]
The Fermi level for sure will be within the gap (at low temperatures), since otherwise the nr. of electrons in the system would be incorrect. For large band gap systems, however, at low temperatures there is a wide range of values for the Fermi-level (within the gap!), which still results in the correct nr. of electrons up to numerical precision.

**Reply by achyuttt** [reply]
Dear Dr. Aradi, thank you for the confirmation. If the fermi level lies within the gap, its numerical value won't in fact matter that much for calculating the bandgap since band gap is  just the difference in energy value around fermi level that has ~0 Density of states. 
Since I could find no literatures that has calculated band gap using Density of states from DFTB theory or DFTB+ software, I just wanted to be sure if the process that I am following is correct or the values given by DFTB+ for my systems are correct.

So, to conclude even for the larger band gap system, the fermi level will be within the gap means that the band gap that we calculate from DOS around that Fermi level would be a correct value.
